lib/tasks/patient_locations.rake

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+namespace :patient_locations do
+  desc "Remove stale patient locations from previous schools"
+  task cleanup_post_school_move: :environment do
+    dry_run = ENV["DRY_RUN"] == "true"
+
+    puts "Starting cleanup of stale patient locations from previous schools"
+    puts "DRY RUN MODE - No records will be destroyed" if dry_run
+
+    locations_by_patient =
+      PatientLocation
+        .where.not(location_id: Location.community_clinic)
+        .where(academic_year: AcademicYear.current)
+        .group_by(&:patient_id)
+
+    patients_with_multiple_locations =
+      locations_by_patient.select { |_, locations| locations.count > 1 }
+
+    puts "Found #{patients_with_multiple_locations.count} patients with multiple locations"
+
+    patient_locations = patients_with_multiple_locations.values.flatten
+
+    progress_bar =
+      ProgressBar.create!(
+        total: patient_locations.size,
+        format: "%a %b\u{15E7}%i %p%% %t",
+        progress_mark: " ",
+        remainder_mark: "\u{FF65}"
+      )
+
+    safe_to_destroy_count = 0
+
+    patient_locations.each do |location|
+      if location.safe_to_destroy?
```

Comment:
I think we need to be careful about this, because some patients may not have been seen by a nurse yet and therefore technically `safe_to_destroy?` but do go to the school and therefore won't want to be destroyed.

We probably need to only destroy a location if it's different to the patient's school.